### PR TITLE
Update documentation for usability

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

Updated README files and CONTRIBUTION.md to increase comprehensibility. 

### Description

In the main README, I clarified the difference between connector options, 'query' and 'target_table_sql'. In the examples README, I removed the need to add a lib folder and JAR file, added a docker compose down instruction, clarified how to run all examples. In CONTRIBUTION.md, I added a troubleshooting step in case the user has not set up SSH key. 

### Related Issue

http://jira.verticacorp.com:8080/jira/browse/VER-77204

### Additional Reviewers

@alexr-bq 
@jonathanl-bq

### CODE REVIEW

**Review by jonathanl-bq — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Git. You should also setup **vertica/spark-connector** as an "upstream" remote.
 git remote add upstream git@github.com:vertica/spark-connector.git
 git fetch upstream
 ```
-
+If you get a `git@github.com: Permission denied (publickey)` error, please follow [this guide](https://stackoverflow.com/questions/58429764/git-could-not-read-from-remote-repository) to set up an SSH key.
```
Comment (alexr-bq) [inline CONTRIBUTING.md:43]: newline after

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -255,3 +258,9 @@ Pull requests are usually reviewed within a few days. If there are comments to a
 That's it! Thank you for your code contribution!
 
 After your pull request is merged, you can safely delete your branch and pull the changes from the upstream repository.
+
+[]: https://stackoverflow.com/questions/58429764/git-could-not-read-from-remote-repository
```
Comment (alexr-bq) [inline CONTRIBUTING.md:262]: what are these?

**Review by alexr-bq — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (ravjotbrar) [reply CONTRIBUTING.md]: Accidental. Removed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ravjotbrar) [reply CONTRIBUTING.md:43]: Done.

**Review by alexr-bq — APPROVED** [review]

**Pull request merged**